rust/src/writer/utils.rs

```diff
@@ -165,3 +193,33 @@ pub(crate) fn stringified_partition_value(
 
     Ok(Some(s))
 }
+
+/// Remove any partition related fields from the schema
+pub(crate) fn schema_without_partitions(
```

Comment:
Also, do you still need this function?

Reply:
Nope. Reverted and restored the original function from where it was sourced.